rust/src/composepost.rs

```diff
@@ -91,19 +84,30 @@ pub(crate) fn compose_init_rootfs(
     Ok(())
 }
 
+/// Prepare rootfs for commit.
+pub fn compose_prepare_rootfs(
+    src_rootfs_dfd: i32,
+    target_rootfs_dfd: i32,
+    treefile: &mut Treefile,
+) -> CxxResult<()> {
+    let src_rootfs = &crate::ffiutil::ffi_view_openat_dir(src_rootfs_dfd);
+    let target_rootfs = &crate::ffiutil::ffi_view_openat_dir(target_rootfs_dfd);
+    compose_prepare_rootfs_impl(src_rootfs, target_rootfs, treefile)?;
+    Ok(())
+}
+
+/// Prepare rootfs for commit.
+///
 /// Initialize a basic root filesystem in @target_root_dfd, then walk over the
 /// root filesystem in @src_rootfs_fd and take the basic content (/usr, /boot, /var)
 /// and cherry pick only specific bits of the rest of the toplevel like compatibility
 /// symlinks (e.g. /lib64 -> /usr/lib64) if they exist.
 #[context("Preparing rootfs for commit")]
-pub(crate) fn compose_prepare_rootfs(
-    src_rootfs_dfd: i32,
-    target_rootfs_dfd: i32,
+fn compose_prepare_rootfs_impl(
+    src_rootfs_dfd: &openat::Dir,
+    target_rootfs_dfd: &openat::Dir,
     treefile: &mut Treefile,
-) -> CxxResult<()> {
```

Comment:
@cgwalters it is my understanding that we can't really mix `CxxResult` and `fn-error-context`, but let me know if I actually misunderstood it.

Reply:
It should work, I added https://github.com/coreos/rpm-ostree/blob/a700b2e74b6bb54c15e6eee4cdbb1559e155301a/rust/src/cxxrsutil.rs#L163 for this.

Reply:
Ah ok, thanks, I hadn't seen that. Let me undo the `_impl` I added here then.